### PR TITLE
Get genesis timestamp from bitcoind

### LINKED ISSUE #904: `startrescan` allows timestamps before genesis block on test networks
Entered a 2015 date as rescan date for a signet setup.
Rescan stopped at 100%, no errors handled by the GUI, only an error in logs:
`2024-01-03T15:09:04.881557Z ERROR liana::bitcoin::poller::looper:289: Could not retrieve block height for timestamp '1421020800'`

### PR DESCRIPTION
**Pull request by pythcoiner**
fixes #904

### CODE REVIEW

File: tests/test_rpc.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 def test_getinfo(lianad):
     res = lianad.rpc.getinfo()
-    assert 'timestamp' in res.keys()
+    assert "timestamp" in res.keys()
```
Comment (pythcoiner) [inline tests/test_rpc.py:25]: black doing is job here

File: tests/test_rpc.py [hunk 2]
```diff
@@ -1232,9 +1234,7 @@ def test_rbfpsbt_cancel(lianad, bitcoind):
     # But we can't set the feerate explicitly.
     with pytest.raises(
         RpcError,
-        match=re.escape(
-            "A feerate must not be provided if creating a cancel."
-        ),
+        match=re.escape("A feerate must not be provided if creating a cancel."),
```
Comment (pythcoiner) [inline tests/test_rpc.py:1237]: black doing his job

**Comment by jp1ac4** [comment]
Is this still WIP?

**Comment by pythcoiner** [comment]
> Is this still WIP?

updated title

File: tests/test_rpc.py [hunk 3]
```diff
@@ -599,7 +599,9 @@ def all_spent(coins):
     with pytest.raises(RpcError, match="Insane timestamp.*"):
         lianad.rpc.startrescan(future_timestamp)
     assert lianad.rpc.getinfo()["rescan_progress"] is None
-    prebitcoin_timestamp = 1231006505 - 1
+    block_hash = bitcoind.rpc.getblockhash(0)
+    genesis_timestamp = bitcoind.rpc.getblock(block_hash)["time"]
+    prebitcoin_timestamp = genesis_timestamp - 1
     with pytest.raises(RpcError, match="Insane timestamp."):
         lianad.rpc.startrescan(prebitcoin_timestamp)
```
Comment (jp1ac4) [inline tests/test_rpc.py:606]: This is not directly related to this PR, but perhaps we should also check that the rescan succeeds if starting from `genesis_timestamp`.

**Review by jp1ac4 — APPROVED** [review]
ACK https://github.com/wizardsardine/liana/pull/957/commits/3eef53de3de9bf720eaed5ed3d49e518a35ee983.

File: tests/test_rpc.py (hunk 3 above)
Comment (pythcoiner) [reply tests/test_rpc.py:606]: adressed

**Review by jp1ac4 — APPROVED** [review]
ACK https://github.com/wizardsardine/liana/commit/f2c418f79a6fe8f1d6138b12af9fdc7303b4725d.
